### PR TITLE
Escape attributes

### PR DESCRIPTION
**Pull request by andy128k**

### CODE REVIEW

File: src/events/attributes.rs [hunk 1]
```diff
@@ -59,13 +59,13 @@ pub struct Attribute<'a> {
     /// the key to uniquely define the attribute
     pub key: &'a [u8],
     /// the value
```
Comment (tafia) [inline src/events/attributes.rs]: Can you mention is the doc that the `value` is the `unescaped` value?

And that converting from a `(&[u8], &[u8])` doesn't try to escape the value while converting from a `(&str, &str)` does?

**Comment by andy128k** [comment]
@tafia Of course, I can make changes in documentation too. But the fact that types of methods "version", "encoding", and "standalone" of BytesDecl are changed disturbs me.

**Comment by tafia** [comment]
> But the fact that types of methods "version", "encoding", and "standalone" of BytesDecl are changed disturbs me.

I agree it is not ideal. On the other hand it make it all more consistent. I don't mind having a `Cow`. I don't really mind having a `&[u8]` either to be honest. One is transparency, the other one is usability (with no performance impact).

**Comment by andy128k** [comment]
Any updates on this PR?

**Comment by tafia** [comment]
Sorry! It completely got under the radar.
I'll have a lot asap!

**Comment by tafia** [comment]
Thanks again for the PR and sorry for the time it took me to merge it!

**Comment by andy128k** [comment]
Thanks!

**Pull request merged**